source/frontend/src/components/common/mapFSM/useMapSearch.tsx

```diff
@@ -132,48 +132,49 @@ export const useMapSearch = () => {
         }
         const planNumberPmbcData = await findByPlanNumberTask;
 
-        // If the property was found on the pims inventory, use that.
-        if (planNumberInventoryData?.features && planNumberInventoryData?.features?.length > 0) {
-          const validFeatures = planNumberInventoryData.features.filter(
-            feature => !!feature?.geometry,
-          );
+        const validFeatures = planNumberInventoryData?.features?.filter(
+          feature => !!feature?.geometry,
+        );
 
-          const attributedFeatures: FeatureCollection<
-            Geometry,
-            PMBC_FullyAttributed_Feature_Properties
-          > =
-            planNumberPmbcData?.features?.length > 0
-              ? {
-                  type: 'FeatureCollection',
-                  features: [...(planNumberPmbcData?.features || [])],
-                  bbox: planNumberPmbcData?.bbox,
-                }
-              : null;
-          const validPmbcFeatures = attributedFeatures.features.filter(
-            feature => !!feature?.geometry,
-          );
+        const attributedFeatures: FeatureCollection<
+          Geometry,
+          PMBC_FullyAttributed_Feature_Properties
+        > =
+          planNumberPmbcData?.features?.length > 0
+            ? {
+                type: 'FeatureCollection',
+                features: [...(planNumberPmbcData?.features || [])],
+                bbox: planNumberPmbcData?.bbox,
+              }
+            : null;
+        const validPmbcFeatures = attributedFeatures?.features?.filter(
+          feature => !!feature?.geometry,
+        );
 
-          result = {
-            pimsLocationFeatures: {
-              type: planNumberInventoryData.type,
-              bbox: planNumberInventoryData.bbox,
-              features: validFeatures,
-            },
-            pimsBoundaryFeatures: emptyPimsBoundaryFeatureCollection,
-            fullyAttributedFeatures: exists(attributedFeatures)
-              ? {
-                  type: attributedFeatures.type,
-                  bbox: attributedFeatures.bbox,
-                  features: validPmbcFeatures,
-                }
-              : emptyPmbcFeatureCollection,
-          };
+        result = {
+          pimsLocationFeatures: exists(validFeatures)
+            ? {
+                type: planNumberInventoryData.type,
+                bbox: planNumberInventoryData.bbox,
+                features: validFeatures,
+              }
+            : emptyPimsLocationFeatureCollection,
+          pimsBoundaryFeatures: emptyPimsBoundaryFeatureCollection,
+          fullyAttributedFeatures: exists(attributedFeatures)
+            ? {
+                type: attributedFeatures.type,
+                bbox: attributedFeatures.bbox,
+                features: validPmbcFeatures,
+              }
+            : emptyPmbcFeatureCollection,
```

Comment:
hmm - this logic seems a bit wonky. On line 150 we calculate **validPmbcFeatures** to be the subset of features with valid geometry (good!). But then here we check that the original **attributedFeatures** exists (not the validPmbcFeatures array) and then we assign the filtered one?

By looking at the pattern for pimsLocationFeature, I would expect that we check for `exists(validPmbFeatures)` on line 163. Am I missing something?